### PR TITLE
Allow more than one extra section in haproxy config

### PR DESCRIPTION
**Pull request by nico-acidtango**
If given more than one extra section only one of these gets to be in the final config. 
I've modified the extra config generation section to allow more than one of these appended at the end sorted by section tag.

### CODE REVIEW

File: provider/haproxy/test_data/custom_config_multiple_extra_resp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+frontend 80
+      bind *:80
+      mode http
+      redirect scheme https code 301 if !{ ssl_fc }
+frontend 81
+      bind *:81
+      mode http
+      redirect scheme https code 301 if !{ ssl_fc }
+listen stats :9000
```
Comment (kinarashah) [inline provider/haproxy/test_data/custom_config_multiple_extra_resp]: Could you change this to [new](https://serverfault.com/questions/792971/haproxy-1-7-statistics-setup/792985)  syntax?  
```
listen stats
  bind :9000
```

**Comment by kinarashah** [comment]
This fix would also solve [rancher/rancher#8226](https://github.com/rancher/rancher/issues/8226). 
Thanks a lot, @nico-acidtango.

**Comment by nico-acidtango** [comment]
Made the suggested changed to the stats syntax. Glad to be of help @kinarashah 😄

**Comment by alena1108** [comment]
@nico-acidtango could you squash your commits into 1?

**Pull request merged**